ArduCopter/mode_auto.cpp

```diff
@@ -2041,4 +2041,32 @@ bool ModeAuto::verify_nav_script_time()
 }
 #endif
 
+/********************************************************************************/
+// Pause and continue auto mode
+/********************************************************************************/
+
+// pause_mission - Prevent aircraft from progressing along the track
+bool ModeAuto::pause()
+{
+    // do not pause if not in the wp_nav sub mode or already reached the wp_nav destination
+    if(!wp_nav->is_active() || wp_nav->reached_wp_destination()) {
+        return false;
+    }
+
+    wp_nav->set_pause();
+    return true;
+}
+
+// continue_mission - Allow aircraft to progress along the track
+bool ModeAuto::resume()
+{
+    // do not resume if not in the wp_nav sub mode or already reached the wp_nav destination
+    if(!wp_nav->is_active() || wp_nav->reached_wp_destination()) {
```

Comment:
I worry that we could get into the state where we cannot call wp_nav->set_continue().  I think it would be safer to remove these checks so the call always succeeds or an an accessor to wp_nav so that this method can check if wp_nav is currently paused or not.  I don't mind which one you choose.. maybe just always allow resume to succeed.

Also I'd prefer if we can keep the names consistent by renaming wp_nav::set_continue() to "resume()".